tensorflow/lite/experimental/shlo/ops/dot_general.cc

```diff
@@ -26,23 +26,126 @@ limitations under the License.
+  absl::InlinedVector<Axis, kMaxNumDimensions> result_dims;
+  for (Axis i = 0; i < rank; ++i) {
+    if (!ContainsDimension(batching_dimensions, i) &&
+        !ContainsDimension(contracting_dimensions, i)) {
+      result_dims.push_back(i);
+    }
+  }
+  return result_dims;
+}
+
+bool CheckZeroPoint(
+    const QuantizedElementTypePerTensor::ZeroPointVariant& zero_point) {
+  return std::visit(
+      [](const auto& v) { return v == static_cast<decltype(v)>(0); },
+      zero_point);
+}
+
+bool CheckZeroPoints(
+    const QuantizedElementTypePerAxis::ZeroPointsVariant& zero_points) {
+  return std::visit(
+      [](const auto& v) {
+        return std::all_of(v.begin(), v.end(), [](const auto value) {
+          return value == static_cast<decltype(value)>(0);
+        });
+      },
+      zero_points);
+}
+
+bool IncrementIndices(const Tensor& lhs,
+                      absl::InlinedVector<Axis, kMaxNumDimensions>& lhs_index,
+                      absl::InlinedVector<Axis, kMaxNumDimensions>& rhs_index,
+                      absl::Span<const Axis> lhs_contracting_dimensions,
+                      absl::Span<const Axis> rhs_contracting_dimensions,
+                      const size_t lhsc_size) {
+  if (lhsc_size == 0) {
+    return false;
+  }
+  for (int64_t i = static_cast<int64_t>(lhsc_size) - 1; i >= 0; --i) {
+    lhs_index[lhs_contracting_dimensions[i]]++;
+    rhs_index[rhs_contracting_dimensions[i]]++;
+    if (lhs_index[lhs_contracting_dimensions[i]] <
+        lhs.shape().Dim(lhs_contracting_dimensions[i])) {
+      return true;
+    }
+    if (i == 0) {
+      return false;
+    }
+    lhs_index[lhs_contracting_dimensions[i]] = 0;
+    rhs_index[rhs_contracting_dimensions[i]] = 0;
+  }
+  return true;
+}
+
 absl::Status CheckParameters(
     const Tensor& lhs, const Tensor& rhs,
-    const absl::Span<int64_t> lhs_batching_dimensions,
-    const absl::Span<int64_t> rhs_batching_dimensions,
-    const absl::Span<int64_t> lhs_contracting_dimensions,
-    const absl::Span<int64_t> rhs_contracting_dimensions, Tensor& output,
-    std::array<PrecisionTypes, 2>& precision_configs) {
-  const DimensionSize lhsb_size = lhs_batching_dimensions.size();
-  const DimensionSize rhsb_size = rhs_batching_dimensions.size();
-  const DimensionSize lhsc_size = lhs_contracting_dimensions.size();
-  const DimensionSize rhsc_size = rhs_contracting_dimensions.size();
+    absl::Span<const Axis> lhs_batching_dimensions,
+    absl::Span<const Axis> rhs_batching_dimensions,
+    absl::Span<const Axis> lhs_contracting_dimensions,
+    absl::Span<const Axis> rhs_contracting_dimensions, Tensor& output,
+    const std::array<PrecisionTypes, 2>& precision_configs) {
+  const size_t lhsb_size = lhs_batching_dimensions.size();
+  const size_t rhsb_size = rhs_batching_dimensions.size();
+  const size_t lhsc_size = lhs_contracting_dimensions.size();
+  const size_t rhsc_size = rhs_contracting_dimensions.size();
   const size_t lhs_rank = lhs.Rank();
```

Comment:
@rascani, `Tensor::Rank()` returns `size_t` type, can we change it to its alias type `Axis` so that it will align with other library functions like `Shape::Dim/Shape::Dims` ?